timescale_vector/client.py

```diff
@@ -431,18 +432,24 @@ def build_query(self, params: List) -> Tuple[str, List]:
                     else:
                         where_conditions.append(f"uuid_timestamp(id) {operator} {param_name}")
                     params.append(value)
-                    continue
-
-                field_cast = ''
-                if isinstance(value, int):
-                    field_cast = '::int'
-                elif isinstance(value, float):
-                    field_cast = '::numeric'
-                elif isinstance(value, datetime):
-                    field_cast = '::timestamptz'   
-
-                where_conditions.append(f"(metadata->>'{field}'){field_cast} {operator} {param_name}")
-                params.append(value) 
+                
+                elif operator == "@>" and (isinstance(value, list) or isinstance(value, tuple)):
+                    if len(value) == 0:
+                        raise ValueError("Invalid value. Empty lists and empty tuples are not supported.")
+                    json_value = json.dumps(value)
+                    where_conditions.append(f"metadata @> jsonb_build_object('{field}', {param_name}::jsonb)")
```

Comment:
Did you verify this is still an indexed operation?

Reply:
yes. it is